broadlink/climate.py

```diff
@@ -268,25 +268,40 @@ def _calculate_checksum(self, payload: bytes, byteorder: str) -> bytes:
         result = (0xffff - s)
         return result.to_bytes(2, byteorder)
 
+    def _encode(self, payload: bytes) -> bytes:
+        """Encode payload i.e. add length to beginning, checksum after payload,
+        and pad size.
+        """
+        import struct
+        length = 2 + len(payload)  # length prefix
+        packet_length = ((length - 1) // 16 + 1) * 16
+        packet = bytearray(2)
+        struct.pack_into("<H", packet, 0, length)
+        packet.extend(payload)
+        packet.extend(self._calculate_checksum(packet[2:], 'little'))
+        packet.extend([0] * (packet_length - len(packet)))  # round size to 16
+
+        return packet
```

Comment:
```python3
    def _encode(self, payload: bytes) -> bytes:
        """Encode the payload."""
        checksum = _calculate_checksum(payload, 'little')
        return struct.pack("H", len(payload) + 2) + payload + checksum
```

Reply:
That's way more compact. Does anything else handle the padding?